usr/src/uts/common/fs/zfs/zfs_vnops.c

```diff
@@ -4424,6 +4429,66 @@ zfs_inactive(vnode_t *vp, cred_t *cr, caller_context_t *ct)
 	rw_exit(&zfsvfs->z_teardown_inactive_lock);
 }
 
+static void
+zfs_inactive_task(void *task_arg)
+{
+	znode_t *zp = task_arg;
+	ASSERT(zp);
+	zfs_inactive_impl(zp, CRED());
```

Comment:
This is called from a taskq, where CRED() == kcred, right?  Is that OK?  It looks like the cred is passed to zfs_putapage(), which ignores it.  I would rather have zfs_inactive_impl() not take a cred, and pass in kcred to pvn_vplist_dirty.